src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -360,7 +360,7 @@ internal BoundExpression BindValueAllowArgList(ExpressionSyntax node, Diagnostic
             out BoundExpression valueBeforeConversion)
         {
             Debug.Assert(this.InParameterDefaultValue);
-            Debug.Assert(this.ContainingMember().Kind == SymbolKind.Method || this.ContainingMember().Kind == SymbolKind.Property);
+            Debug.Assert(this.ContainingMemberOrLambda.Kind == SymbolKind.Method || this.ContainingMemberOrLambda.Kind == SymbolKind.Property);
```

Comment:
I didn't understand the second part of the condition (`this.ContainingMemberOrLambda.Kind == SymbolKind.Property`).
How can we get a lambda that is a property?